### PR TITLE
docs(dynamic-links): add set up docs

### PR DESCRIPTION
**Pull request by russellwheatley**
- setup docs for dynamic links
- update `firebase.app` documentation example.
- added note that realtime database ordering is respected when you use `forEach` method.

### CODE REVIEW

**Comment by mikehardy** [comment]
I have the most undirected feedback ever (I truly apologize) but the dynamic links area is the last area that seems to be a real disaster w.r.t. user support.

I've been tracking (but not really participating in) the discussions on a couple issues, they are either illuminating or really confusing depending probably on how much experience you have with dynamic links

https://github.com/invertase/react-native-firebase/issues/1901#issuecomment-563828725

This one seems really nasty but really important (the facebook share case, basically) https://github.com/invertase/react-native-firebase/issues/3485 and https://github.com/firebase/firebase-android-sdk/issues/916

Not sure if this still applies https://github.com/invertase/react-native-firebase/issues/3249
Bizarre corner case fail https://github.com/invertase/react-native-firebase/issues/3261

Probably still a problem based on other issues mentioning post-store-install issues https://github.com/invertase/react-native-firebase/issues/1311 e.g. https://github.com/invertase/react-native-firebase/issues/2661

A building link issue on android? https://github.com/invertase/react-native-firebase/issues/3086

Probably project-specific but indicator docs don't work for everyone https://github.com/invertase/react-native-firebase/issues/2660

Crash with patch-package workaround? https://github.com/invertase/react-native-firebase/issues/2807

Some sort of data caching issue with multiple dynamic links? https://github.com/invertase/react-native-firebase/issues/2631 (and https://github.com/invertase/react-native-firebase/issues/3094)

That is obviously a comment bomb but it at the very least gets them out of a mailbox I've been tracking them in and puts them all in one spot here where at worst I can personally refer to them. If scanning them helps point out any dynamic link issues that need doc fixes (or crash fixes + data caching fixes) that'd be good too :)

File: docs/dynamic-links/usage/index.md [hunk 1]
```diff
@@ -36,6 +36,53 @@ a Dynamic Link on iOS or Android, they can be taken directly to the linked conte
...
+4. Open your project in Xcode and open your app under the `TARGETS` header. Click the `Signing & Capabilities` tab. You will need to ensure your `Team` is registered, and your `Provisioning Profile` field is completed. Please add the domain you created in your
+   Firebase console to the `Associated Domains` and prefix with `applinks:`
+
+![ios dynamic link fourth step](https://images.prismic.io/invertase/d7a8696f-bef1-44a4-b048-aab7bb534f71_Screenshot+2020-05-07+at+09.55.53.png?auto=compress,format)
+
+5. Click the `Info` tab, and add a `URL Type` to your project. The `Identifier` can be called `Bundle Id` or whatever you wish. Add your bundle identifier to the `URL Schemes` property.
+
+![ios dynamic link fifth step](https://images.prismic.io/invertase/cb029ba6-ad40-494e-a3f6-2aacaff494d1_Screenshot+2020-05-07+at+10.16.16.png?auto=compress,format)
+
+6. To test the dynamic link works, you will need to use a real device as it will not work on a simulator. The notes app is a good place to paste you link and test it opens your app (ensure the app is installed on the device).
```
Comment (cvharris) [inline docs/dynamic-links/usage/index.md:73]: Is this to say FB dynamic links don't work on simulators? Is that particular to RNFB or to the Firebase SDK in general?

Comment (Salakar) [reply docs/dynamic-links/usage/index.md:73]: I think it is possible to test it on the simulator, @russellwheatley you may have accidentally disabled it on the simulator, see: https://youtu.be/KLBjAg6HvG0?t=500

Comment (Salakar) [reply docs/dynamic-links/usage/index.md:73]: Those points on those video might be good to add to troubleshooting links not opening on iOS section? And also documenting how to test the installed & non-installed flows on iOS - see https://stackoverflow.com/questions/39605657/how-can-i-test-firebase-dynamic-links-if-my-app-is-not-in-the-app-store that summarizes it well.

Comment (russellwheatley) [reply docs/dynamic-links/usage/index.md:73]: If you listen from 7 minutes on, Todd says that Dynamic Links do not work on the simulator: https://www.youtube.com/watch?v=KLBjAg6HvG0&feature=youtu.be&t=500. The documentation is based on iOS screencast pt.1. I can add documentation on how to test without the app being installed. I'll also watch iOS screencast pt.2 to see if there's anything else that could be added to the documentation.

Comment (Salakar) [reply docs/dynamic-links/usage/index.md:73]: @russellwheatley fair enough re the simulator - be good to document those troubleshooting steps still I think

**Comment by russellwheatley** [comment]
The iOS pt.2 video seemed out of scope for a "How To Do" guide, it was more of a look into how to handle dynamic link params within Swift code.

**Pull request merged**